### ISSUE TITLE
[FEATURE] Allow movement settings on a per-ability instead of per-action basis

### ISSUE THREAD

**Issue by fwosar** [Hekili/hekili#3100]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/enhancement) for this request and was not able to find one.
- [X] I edited the title of this feature request (above) so that it describes the issue I am reporting.

### Feature Request

Currently, Hekili will recommend abilities like ground targetted abilities or even abilities that have a cast time while moving (for example, Chi Burst). Specifically during M+, where you often pull an extended number of groups, this leads to recommendations at times when you really wouldn't want to cast a specific ability because the mobs will no longer be at that location mere seconds later or because you need to keep up. For some classes, you built optional exceptions, like Aimed Shot in marksman for example, and for other cases you can set a movement on a per-action basis. However, I think it would be a lot more intuitive and easier to add a per-ability setting as well. The per-ability setting could be used as the default for all priorities that don't have an explicit movement condition set, while still allowing a per-action override.

This would make it way easier to tune the action lists without actually having to edit them and having to reapply those edits on each update. This setup would also be flexible enough to model even more complex movement scenarios because it is easy to add exceptions to the baseline. I really hope you consider this, as editing the offending abilities in the Windwalker action lists, of which there are roughly 20, after each update really becomes annoying after a while.

**Comment by Hekili** [comment]
This is something I've been considering, but the UI would have to be improved to clarify that something is(n't) recommended due to movement based on ability settings vs. priority, etc.

**Comment by designadrug** [comment]
I was going to submit a suggestion that when moving, Hekili only suggests abilities that can be cast while moving. 
With the original suggestion above, you could always just drop those abilities with cast-impact times.

**Comment by Hekili** [comment]
> I was going to submit a suggestion that when moving, Hekili only suggests abilities that can be cast while moving.
> With the original suggestion above, you could always just drop those abilities with cast-impact times.

This requires that priorities are written to do "what is optimal when moving" vs. "what is optimal when not moving" which is not the case consistently across classes/specializations.

In most cases, the thing you should do when moving is to stop so you can cast the recommended ability, assuming you won't die.

**Comment by johnnylam88** [comment]
I agree that it's better if the priority itself had a `movement` action list that was run if moving. For Ovale, I had to make a similar decision and decided that it was better just to follow the priority instead of casting only a handful of instant-cast spells.
